src/frontend/admin/src/components/ConfigurationEditor.tsx

```diff
@@ -119,19 +132,51 @@ function AdvancedJson(): React.ReactElement {
         <Container>
             <Container maxWidth="md" sx={{ pt: 2 }}>
                 <Box height="75vh">
-                    <JsonEditor
+                    {editorType === EditorLanguage.Json && <JsonEditor
                         schema={schema}
                         defaultValue={content}
                         onChange={(value: string) => setContent(value)}
-                    />
+                    />}
+                    {editorType === EditorLanguage.CSV && <CsvCodeEditor
+                        defaultValue={content}
+                        onChange={(value: string) => setContent(value)}
+                    />}
                 </Box>
                 <Button type="submit" onClick={onSubmit} sx={{ mt: 2 }}>
                     Save
                 </Button>
             </Container>
-            <ExamplesContainer />
+            <ExamplesContainer apiRoot={apiRoot}/>
         </Container>
     );
 }
 
-export default AdvancedJson;
+enum EditorType {
+    advancedJson = "advancedJson",
+    visualConfig = "visualConfig",
+    customFields = "customFields",
+}
+
+function ConfigurationsEditor() : React.ReactElement {
```

Comment:
nitpick: I wonder if we should just move it to a separate file at that point, but don't have too many opinions.